unified/install.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+#
+# Copyright (C) 2020 ScyllaDB
+#
+
+#
+# This file is part of Scylla.
+#
+# Scylla is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Scylla is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+set -e
+
+print_usage() {
+    cat <<EOF
+Usage: install.sh [options]
+
+Options:
+  --root /path/to/root     alternative install root (default /)
+  --prefix /prefix         directory prefix (default /usr)
+  --python3 /opt/python3   path of the python3 interpreter relative to install root (default /opt/scylladb/python3/bin/python3)
+  --housekeeping           enable housekeeping service
+  --nonroot                install Scylla without required root priviledge
+  --sysconfdir /etc/sysconfig   specify sysconfig directory name
+  --help                   this helpful message
+EOF
+    exit 1
+}
+
+root=/
+housekeeping=false
+python3=/opt/scylladb/python3/bin/python3
+sysconfdir=/etc/sysconfig
+nonroot=false
+
+while [ $# -gt 0 ]; do
+    case "$1" in
+        "--root")
+            root="$2"
+            shift 2
+            ;;
+        "--prefix")
+            prefix="$2"
+            shift 2
+            ;;
+        "--housekeeping")
+            housekeeping=true
+            shift 1
+            ;;
+        "--python3")
+            python3="$2"
+            shift 2
+            ;;
+        "--nonroot")
+            nonroot=true
+            shift 1
+            ;;
+        "--sysconfdir")
+            sysconfdir="$2"
+            shift 2
+            ;;
+        "--help")
+            shift 1
+	    print_usage
+            ;;
+        *)
+            print_usage
+            ;;
+    esac
+done
+
+if [ -z "$prefix" ]; then
+    if $nonroot; then
+        prefix=~/scylladb
+    else
+        prefix=/opt/scylladb
+    fi
+fi
+
+if $housekeeping; then
+    arg_housekeeping=--housekeeping
+fi
+if $nonroot; then
+    arg_nonroot=--nonroot
+fi
+
+(cd scylla; ./install.sh --root "$root" --prefix "$prefix" --python3 "$python3" "$arg_housekeeping" "$arg_nonroot" --sysconfdir "$sysconfdir")
```

Comment:
"$arg_housekeeping" translates to an empty argument instead of nothing.

I think it's nicer to use arrays:

```sh
args=()

if $housekeeping; then
    args+=(--housekeeping)
fi

if $nonroot; then
    args+=(--nonroot)
fi
```

Reply:
@avikivity but we only need --housekeeping for scylla main package, but --nonroot needed for all package.

Reply:
Does it work when $arg_housekeeping or $arg_nonroot is empty? I think it generates an empty parameter that will confuse install.sh.